resolves #1338 - adds hreflang metadata

ChrisBAshton opened the pull request:
Resolves #1338

**Overall change:** Adds three new metadata values. The `.com` variant which has `en`. One is a specific Great Britain version and required the creation of a link which always ends in `.co.uk`.
The x-default is needed for SEO purposes to indicate which hrefLang is the default.

**Code changes:**

- Adds alternateLinks to Metadata component. 
- Adds 3 metadata links to the document head for News. Not necessary for Persian. 
- Updates snapshots

**Testing notes**
- Persian canonical and amp pages **should not** have alternate links: 
- http://localhost:7080/persian/articles/cyddjz5058wo
- http://localhost:7080/persian/articles/cyddjz5058wo.amp
- News canonical and amp pages **should** have alternate links: 
- http://localhost:7080/news/articles/cl55zn0w0l4o
- http://localhost:7080/news/articles/cl55zn0w0l4o.amp

<img width="297" alt="Screen Shot AMP" src="https://user-images.githubusercontent.com/3028997/55966730-dd46ad80-5c70-11e9-9030-2e4e9ff123b3.png">
<img width="300" alt="Screen Shot Canonical" src="https://user-images.githubusercontent.com/3028997/55966732-dd46ad80-5c70-11e9-823c-cf59570bb6a9.png">

---

- [x] I have assigned myself to this PR and the corresponding issues
- [ ] Tests added for new features
- [ ] Test engineer approval
- [ ] I have followed [the merging checklist](https://github.com/bbc/simorgh-infrastructure/blob/latest/documentation/MERGE_PROCESS.md) and this is ready to merge.

ChrisBAshton committed:
resolves #1338 - adds hreflang metadata

dr3 commented on src/app/components/Metadata/index.jsx:
```diff
@@ -16,6 +16,7 @@ const Metadata = ({
   articleSection,
   brandName,
   canonicalLink,
+  canonicalLinkGb,
```
Surely we need to actually pass this from the container?

ChrisBAshton replied:
🤦‍♂️

ChrisBAshton committed:
adds canonicalLinkGb to container

sareh commented on src/app/components/Metadata/index.jsx:
```diff
@@ -92,6 +96,7 @@ Metadata.propTypes = {
   articleSection: string,
   brandName: string.isRequired,
   canonicalLink: string.isRequired,
+  canonicalLinkGb: string.isRequired,
```
This works for this specific English language page, however it doesn't scale well to other sites.

We should pass an array of alternateLinks to this front-end component, each with a hrefLang and href value. This array should be looped over for this component. The values should be set in the Metadata Container. 

```
const alternateLinks = [
  {
    hrefLang: 'x-default',
    href: 'https://www.bbc.com/news/articles/c0000000001',
  }
  {
    hrefLang: 'en',
    href: 'https://www.bbc.com/news/articles/c0000000001',
  },
  {
    hrefLang: 'en-gb',
    href: 'https://www.bbc.co.uk/news/articles/c0000000001',
  },
];
```

sareh committed:
Merge branch 'latest' into hreflang-metadata

sareh committed:
Update to use alternateLinks prop so that alternate <link> elements can vary per service

sareh commented on src/app/containers/Metadata/index.jsx:
```diff
@@ -61,6 +62,20 @@ const MetadataContainer = ({ metadata, promo }) => {
                 />
                 <Metadata
                   isAmp={platform === 'amp'}
+                  alternateLinks={[
```
This sets the alternate links to be `en` and `en-gb` for all services. A later PR will be needed to generalise this for all non-English services. This array of values will then need to live in the `ServiceContext`.

sareh commented:
The codeclimate change suggestions will lead to test files that are less readable by humans, so I suggest disregarding the suggestions for this PR.

sareh committed:
Merge branch 'latest' into hreflang-metadata

sareh committed:
Update alternateLinks logic to handle amp, canonical for services in english and otherwise.

sareh committed:
Add key for mapping through alternateLinks

Clarify it is services with English as their main language that need these alternatelinks

twinlensreflex reviewed (dismissed):
Looks good to me 👍 how did you get `rel=alternate` links in the same place in the end?

sareh committed:
Merge branch 'latest' into hreflang-metadata

pjlee11 reviewed (dismissed):
LGTM 👍

sareh committed:
Remove unneccessary canonicalLinkGb

sareh committed:
Remove unneccessary canonicalLinkGb

sareh committed:
Remove unneccessary canonicalLinkGb

oluoluoxenfree reviewed (approved):
LGTM :)

sareh committed:
Merge branch 'latest' into hreflang-metadata
